src/main/applicationRunner.ts

```diff
@@ -0,0 +1,44 @@
+import * as express from 'express'
+import * as fs from 'fs'
+import * as path from 'path'
+import * as https from 'https'
+
+const logger = require('@hmcts/nodejs-logging').getLogger('applicationRunner')
+
+export class ApplicationRunner {
+  static run (app: express.Application): void {
+    const port = ApplicationRunner.applicationPort()
+    if (app.locals.ENV === 'development' || app.locals.ENV === 'dockertests') {
+      const server = https.createServer(ApplicationRunner.getSSLOptions(), app)
+      server.listen(port, () => {
+        logger.info(`Listener started (PID ${process.pid}): https://localhost:${port}`)
+      })
+    } else {
+      app.listen(port, () => {
+        logger.info(`Listener started (PID ${process.pid}): http://localhost:${port}`)
+      })
+    }
+  }
+
+  private static getSSLOptions (): any {
+    const sslDirectory = path.join(__dirname, 'resources', 'localhost-ssl')
+    return {
+      key: fs.readFileSync(path.join(sslDirectory, 'localhost.key')),
+      cert: fs.readFileSync(path.join(sslDirectory, 'localhost.crt'))
+    }
+  }
+
+  /**
+   * Return type is number or string because Azure PaaS communicates with the application
+   * through a named pipe and not a TCP port.
+   */
+  private static applicationPort (): number | string {
+    const defaultPort = 3000
+    let port: number | string = process.env.PORT
```

Comment:
this won't return a number its always a string from an env variable

Reply:
Fair point, will update.